docs/specs/NETWORKING.md

```diff
@@ -1,3 +1,769 @@
+
+Libp2p's Kad DHT will allow advertising and finding peers by multiple keys, e.g. topics/subnets.
+As `ENR` has a size limit (`< 300` bytes), and therefore discv5 won't support multiple key/value pairs.
+
+Notes:
+- discv5 specifies [Topic Index](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-rationale.md#the-topic-index) that help to lookup relevant nodes in a smaller set of nodes, currently not fully implemented
+- See [Ethereum specs > phase 0 > p2p interface](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#why-are-we-using-discv5-and-not-libp2p-kademlia-dht) 
+details why discv5 was chosen over libp2p Kad DHT in Ethereum.
+
+In `v0` discv5 is used, `v1` TBD, this section will be updated once that work is complete.
+
+### Network ID
+
+Network ID is a `32byte` key, that is used to distinguish between other networks (ssv and others).
+Peers from other public/private libp2p networks (with different network ID) won't be able to read or write messages in the network, 
+meaning that the key to be known and used by all network members.
+
+It is done with [libp2p's private network](https://github.com/libp2p/specs/blob/master/pnet/Private-Networks-PSK-V1.md),
+which encrypts/decrypts all traffic with the corresponding key,
+regardless of the regular transport security protocol ([go-libp2p-noise](https://github.com/libp2p/go-libp2p-noise)).
+
+**NOTE** discovery communication (UDP) won't use the network ID, 
+and it's not needed as unknown peers will be filtered.
+
+
+### Subnets
+
+Consensus messages are being sent in the network over a subnet (pubsub topic), which the relevant peers should be subscribed to.
+
+#### Subnets - fork v0
+
+`v0` was working with a simple approach, each validator committee has 
+a dedicated pubsub topic with all the relevant peers subscribed to it (committee + exporter). \
+It helps to reduce amount of messages in the network, 
+but increases the number of topics which will grow up to the number of validators.
+
+In order to provide more redundancy, there is a global topic (AKA `main topic`) to publish all the decided messages in the network.
+
+#### Subnet - fork v1
+
+A subnet of operators is responsible for multiple committees,
+reusing the same topic to communicate on behalf of multiple validators.
+
+In comparison to `v0`, the number of topics will be reduced and the number of messages sent over the network should grow. \
+As messages will be propagated to a larger set of nodes, we can expect better reliability (arrival of messages to all operators in the committee).
+
+In addition, a larger group of operators provides:
+- redundancy of decided messages across multiple nodes
+- better security for subnets as more nodes will validate messages and can score bad/malicious nodes that will be pruned accordingly.
+
+Global/main topic is not needed in this solution and therefore will be reduced.
+
+**Validators Mapping**
```

Comment:
Not sure where we should write it but having this mapping means that as an operator node, regardless of my shares, I'll save highest decided (and potentially historical decided if flag is set) for all the subnets I'm in at